### DISCUSSION TITLE
Try to implement bidirectional Dshot: requesting help for electronics

### DISCUSSION THREAD

**Discussion by AdrienGuenard** [bitcraze/discussions#1673] [General]
Hi all,
I try to implement bidirectionnal dshot in Bolt1.1 firmware and I would appreciate some help on electronics. This is were I am: 
The Bolt is connected to a 4-in-1 ESC (compatible with bidirectionnel dshot, firmware AM32), motor pin set to output push-pull, then sends a bidirectional dshot frame, and set pin to input. The ESC responds with a frame on the same line, however the voltage level is not the same (see picture of oscilloscope). The GND and VCC on ESC and Bolt are connected, has anyone an idea? Should it be due to a resistance on Bolt? (I don't have this problem with Betaflight controler replacing Bolt)
![image](https://github.com/user-attachments/assets/473443db-f6ff-4aaf-969a-381547f5f860)

**Comment by tobbeanton** [comment]
My initial guess is that the STM32 is not releasing (putting them in HI-Z) the pins after it has sent the DSHOT frame and preventing the ESC from pulling them low.

**Reply by AdrienGuenard** [reply]
Thanks, I will check this
